integration_tests/e2e/prisonerList/prisonerList.cy.ts

```diff
@@ -36,48 +35,121 @@ context(`Display the prisoner list screen`, () => {
+  describe('filtering', () => {
+    it('should filter for non existent prisoner and get zero results', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
 
-    // When
-    prisonerListPage //
-      .setNameFilter('some non existent search term')
-      .applyFilters()
+      // When
+      prisonerListPage //
+        .setNameFilter('some non existent search term')
+        .applyFilters()
 
-    // Then
-    prisonerListPage.hasNoResultsDisplayed()
+      // Then
+      prisonerListPage.hasNoResultsDisplayed()
+    })
+
+    it('should filter for prisoners named John', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
+
+      const numberOfPrisonersNamedJohn = prisonerSearchSummaries.filter(
+        prisonerSearchSummary => prisonerSearchSummary.firstName === 'JOHN',
+      ).length
+
+      // When
+      prisonerListPage //
+        .setNameFilter('John')
+        .applyFilters()
+
+      // Then
+      prisonerListPage //
+        .hasResultsDisplayed(numberOfPrisonersNamedJohn)
+    })
+
+    it('should clear filters to reset the search', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
+      prisonerListPage //
+        .setNameFilter('some non existent search term')
+        .setStatusFilter('NEEDS_PLAN')
+        .applyFilters()
+        .hasNoResultsDisplayed()
+
+      const expectedResultCount = prisonerSearchSummaries.length
+
+      // When
+      prisonerListPage.clearFilters()
+
+      // Then
+      prisonerListPage //
+        .hasResultsDisplayed(expectedResultCount)
+        .hasNoSearchTerm()
+        .hasNoStatusFilter()
+    })
   })
 
-  it('should clear filters to reset the search', () => {
-    // Given
-    cy.signIn()
-    cy.visit('/')
-    const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
-    prisonerListPage //
-      .setNameFilter('some non existent search term')
-      .setStatusFilter('NEEDS_PLAN')
-      .applyFilters()
-      .hasNoResultsDisplayed()
+  describe('sorting', () => {
+    it('should be sorted by reception-date descending as the default sort order', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
 
-    // When
-    prisonerListPage.clearFilters()
+      // When
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
 
-    // Then
-    prisonerListPage //
-      .hasResultsDisplayed()
-      .hasNoSearchTerm()
-      .hasNoStatusFilter()
+      // Then
+      prisonerListPage //
+        .isSortedBy('reception-date', 'descending')
+    })
+
+    it('should sort by location descending', () => {
+      // Given
+      cy.signIn()
+      cy.visit('/')
+      const prisonerListPage = Page.verifyOnPage(PrisonerListPage)
+
+      const expectedFirstRowInTable = prisonerSearchSummaries
```

Comment:
As with the "filtering by John" test above, given that `prisonerSearchSummaries` was randomly generated we need to sort it in the test here to work out what the first row should be